### ISSUE TITLE
linter: `import/namespace` reports an existing export as missing

### ISSUE THREAD

**Issue by afgomez** [oxc-project/oxc#13258]
### What version of Oxlint are you using?

1.12.0

### What command did you run?

`oxlint`

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
  "$schema": "./node_modules/oxlint/configuration_schema.json",
  "plugins": [
    "import",
  ],
  "categories": {
    "correctness": "error"
  },
  "rules": {
  },
  "settings": {
  },
  "env": {
    "builtin": true
  },
  "globals": {},
  "ignorePatterns": []
}
```

### What happened?

The `import/namespace` rule incorrectly reports an existing exported type as missing. When running the linter with the following snippet, the linter reports an error on the second use of `z.infer`.

```ts
import * as z from "zod"; // Using version ^3.25.76

const foo = z.object({
  foo: z.string(),
});

// No error reported in this line
export type Foo = z.infer<typeof foo>;

// × eslint-plugin-import(namespace): "infer" not found in imported namespace "zod".
export interface Bar extends z.infer<typeof foo> {
  bar: string;
}
```

**Comment by camc314** [comment]
@Boshen this is happening because we are getting exports for the JS file rather than the type declaration file.

I'd prefer to fix this in the resolver (and get the declaration file rather than the JS file), what do you think?

**Comment by bastienlimbour** [comment]
I think I am hitting a related case with `@rn-primitives/*@1.4.0`.

Example:

```tsx
import * as SeparatorPrimitive from "@rn-primitives/separator";

function Example() {
  return <SeparatorPrimitive.Root />;
}
```

TypeScript resolves this correctly and `tsc --noEmit` passes.

But oxlint reports:

```text
"Root" not found in imported namespace "@rn-primitives/separator". import(namespace)
```

This also happens with:

```text
@rn-primitives/checkbox: Root, Indicator
@rn-primitives/label: Root, Text
```

The `.d.ts` files expose the symbols correctly:

```ts
// dist/index.d.ts
export * from './separator';

// dist/separator.d.ts
export { Root };
```

But the JS/ESM entrypoint is:

```js
// dist/index.mjs
export * from "./separator";
```

And the target JS file preserves JSX:

```jsx
// dist/separator.mjs
return <Component ... />;
```

So this seems aligned with the comment above: oxlint is resolving/analyzing the JS entrypoint instead of relying on the type declaration file, which produces a false positive for a package that TypeScript accepts.

**Comment by diegohaz** [comment]
Confirmed this is still reproducible with Oxlint 1.73.0. Here’s a generic minimal reproduction:

https://github.com/diegohaz/oxlint-import-namespace-repro

The fixture package exposes a runtime value from `index.js` and a type-only interface from `index.d.ts`:

```ts
import { helper } from "./helper.js";
import * as library from "fixture-package";

helper();
console.log(library.runtimeValue);

export interface Options extends library.PublicOptions {}
```

`npm run typecheck` passes, but `npm run compare` produces:

```text
Oxlint 1.72.0
exit code: 0

Oxlint 1.73.0
index.ts:7:42: error import(namespace): "PublicOptions" not found in imported namespace "fixture-package".
exit code: 1
```

This was previously masked by the control-flow bug fixed in [#24094](https://github.com/oxc-project/oxc/pull/24094). In 1.72.0, an earlier named import caused the rule to stop processing import entries before reaching the namespace import. In 1.73.0, the rule continues processing and exposes the older declaration-resolution false positive described here.

I’d be happy to turn this reproduction into a failing test if useful.
